server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -0,0 +1,265 @@
+import org.opensearch.action.admin.cluster.configuration.ClearVotingConfigExclusionsRequest;
+import org.opensearch.action.admin.cluster.configuration.ClearVotingConfigExclusionsResponse;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.ClusterStateObserver;
+import org.opensearch.cluster.ClusterStateTaskConfig;
+import org.opensearch.cluster.ClusterStateTaskListener;
+import org.opensearch.cluster.ClusterStateUpdateTask;
+import org.opensearch.cluster.coordination.NodeRemovalClusterStateTaskExecutor;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.allocation.AllocationService;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.Priority;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Helper controller class to remove list of nodes from the cluster and update status
+ *
+ * @opensearch.internal
+ */
+
+public class DecommissionController {
+
+    private static final Logger logger = LogManager.getLogger(DecommissionController.class);
+
+    private final NodeRemovalClusterStateTaskExecutor nodeRemovalExecutor;
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final ThreadPool threadPool;
+
+    DecommissionController(
+        ClusterService clusterService,
+        TransportService transportService,
+        AllocationService allocationService,
+        ThreadPool threadPool
+    ) {
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        this.nodeRemovalExecutor = new NodeRemovalClusterStateTaskExecutor(allocationService, logger);
+        this.threadPool = threadPool;
+    }
+
+    /**
+     * Transport call to add nodes to voting config exclusion
+     *
+     * @param nodes set of nodes Ids to be added to voting config exclusion list
+     * @param listener callback for response or failure
+     */
+    public void excludeDecommissionedNodesFromVotingConfig(Set<String> nodes, ActionListener<Void> listener) {
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            AddVotingConfigExclusionsAction.NAME,
+            new AddVotingConfigExclusionsRequest(
+                Strings.EMPTY_ARRAY,
+                nodes.toArray(String[]::new),
+                Strings.EMPTY_ARRAY,
+                TimeValue.timeValueSeconds(30)
```

Comment:
nit: lets use a larger timeout since its possible that the cluster is unresponsive when the decommission might actually be needed

Reply:
Keeping it at 120sec as discussed